doc/TAEF.md

```diff
@@ -1,9 +1,28 @@
-### TAEF ###
+### TAEF Overview ###
+
 TAEF, the Test Authoring and Execution Framework, is used extensively within the Windows organization to test the operating system code in a unified manner for system, driver, and application code. As the console is a Windows OS Component, we strive to continue using the same system such that tests can be ran in a unified manner both externally to Microsoft as well as inside the official OS Build/Test system.
 
-The [official documentation](https://msdn.microsoft.com/en-us/library/windows/hardware/hh439725\(v=vs.85\).aspx) for TAEF describes the basic architecture, usage, and functionality of the test system. It is similar to Visual Studio test, but a bit more comprehensive and flexible.
+The [official documentation](https://docs.microsoft.com/en-us/windows-hardware/drivers/taef/) for TAEF describes the basic architecture, usage, and functionality of the test system. It is similar to Visual Studio test, but a bit more comprehensive and flexible.

```

Comment:
I gotta say I'm surprised that this file hasn't changed since "Initial release of the Windows Terminal source code". That's 17 months! This link to TAEF doc hasn't been working for god knows how long, and no one to my knowledge actually complaines about it?...

Reply:
I mean, the core contributors are always aware of how to run it without reading the doc. And the CI will crack anyone else over the head on their PR if they break a test. Also the top search engine result for something like 'TAEF usage' leads to the right place... so I can understand how it decayed. But glad you fixed it. :)

Reply:
Well google helps a lot, there’s no doubt about it. But it does pose certain challenges to newcomers of this repo to realize how to even getting started with unit testing. It took me some time to even realize what is the keyword to google. I’m sure @german-one feels the same.

Reply:
Yes, that has been the main reason why I once filed the issue. If contributors are asked to write a unit test for their code for the first time they will likely get a problem to find the related documentation. If you look into existing unit tests you'll only find `WEX` rather than `TAEF`. Try to google WEX and you'll see how long it may take until you realize that TAEF is related to it and is the key to find the necessary information 😀

Reply:
Speaking of search engine, I use Bing a lot. Like really a lot. So no hard feelings for anyone in the Bing team 😆